clients/ui/frontend/src/shared/context/NamespaceSelectorContext.tsx

```diff
@@ -40,6 +48,24 @@ const EnabledNamespaceSelectorContextProvider: React.FC<NamespaceSelectorContext
 
   const firstNamespace = namespaces.length > 0 ? namespaces[0] : null;
 
+  React.useEffect(() => {
+    if (isIntegrated()) {
+      // Initialize the central dashboard client
+      try {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        window.centraldashboard.CentralDashboardEventHandler.init((cdeh: any) => {
+          // eslint-disable-next-line no-param-reassign
+          cdeh.onNamespaceSelected = (newNamespace: string) => {
+            setPreferredNamespace({ name: newNamespace });
+          };
+        });
+      } catch (err) {
+        /* eslint-disable no-console */
+        console.error('Failed to initialize central dashboard client', err);
+      }
+    }
+  }, []);
```

Comment:
Is it ok for us to use no dependencies here?

Reply:
There's no external dependency that will re-render, that's why our linter don't catch anything, this is used to initialization in the first render of the context.